File: java-example/src/main/java/kclexample/java/ExampleRecordProcessorFactory.java
package kclexample.java;

import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessor;
import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory;

public class ExampleRecordProcessorFactory implements IRecordProcessorFactory {

    private final String tableName;

    ExampleRecordProcessorFactory(String tableName) {
        this.tableName = tableName;
    }

    @Override
    public IRecordProcessor createProcessor() {
        return new ExampleRecordProcessor(tableName);
    }
}

File: kotlin-example/src/main/kotlin/kclexample/kotlin/ExampleRecordProcessorFactory.kt
package kclexample.kotlin

import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessor
import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory

class ExampleRecordProcessorFactory(
        private val tableName: String
): IRecordProcessorFactory {

    override fun createProcessor(): IRecordProcessor {
        return ExampleRecordProcessor(tableName)
    }
}
